src/mmw/js/src/core/views.js

```diff
@@ -808,24 +812,94 @@ var MapView = Marionette.ItemView.extend({
-        this._dataCatalogResultsLayer.eachLayer(function(layer) {
-            var result = resultModels.findWhere({ id: layer.options.id });
-            layer.on('popupopen', function() {
-                layer.setStyle(dataCatalogActiveStyle);
-                result.set('active', true);
-            });
-            layer.on('popupclose', function() {
-                layer.setStyle(dataCatalogPointStyle);
-                result.set('active', false);
-            });
-            layer.bindPopup(new PopoverView({
-                    model: result,
-                    catalog: catalogId
-                }).render().el, { className: 'data-catalog-popover' });
-        });
+    bindDataCatalogPopovers: function(SinglePopoverView, ListPopoverView,
+                                      catalogId, resultModels) {
+        var self = this,
+            handleClick = function(e) {
+                var clickPoint = e.layerPoint,
+                    clickLatLng = e.latlng,
+
+                    intersectsClickBounds = function(layer) {
+                        var shape = layer.getLayers()[0];
+
+                        if (shape instanceof L.Polygon) {
+                            return shape.getBounds().contains(clickLatLng);
+                        }
+
+                        if (shape instanceof L.Circle) {
+                            return shape._point.distanceTo(clickPoint) <= shape._radius;
+                        }
+
+                        return false;
+                    },
+
+                    // Get a list of results intersecting the clicked point
+                    intersectingFeatures = _.reduce(
+                        e.target._layers,
+                        function(intersectingFeatures, layer) {
+                            if (intersectsClickBounds(layer)) {
+                                intersectingFeatures.push(layer);
+                            }
+                            return intersectingFeatures;
+                        }, []);
+
+                // If nothing intersected the clicked point, finish
+                if (intersectingFeatures.length === 0) {
+                    return;
+                }
+
+                // If only a single feature intersected the clicked point
+                // show its detail popup, put active styling on the feature
+                if (intersectingFeatures.length === 1) {
+                    var layer = intersectingFeatures[0],
+                        result = resultModels.findWhere({ id: layer.options.id });
+
+                    layer.bindPopup(new SinglePopoverView({
+                        model: result,
+                        catalog: catalogId
+                    }).render().el, { className: 'data-catalog-popover'});
+
+                    layer.openPopup();
+                    layer.setStyle(dataCatalogActiveStyle);
+                    result.set('active', true);
+
+                    layer.on('popupclose', function() {
```

Comment:
Is this event handler taken off at any point? Or do they just pile on with more and more clicks?

Reply:
They were probably piling, good catch. I'm changing this to `once`